spec/unit/trackable_spec.rb

```diff
@@ -412,7 +412,7 @@ class CustomTracker < MyModel
             track_history on: :key, changes_method: :my_changes, track_create: true
 
             def my_changes
-              changes.merge('key' => "Save history-#{key}")
+              changes.merge('key' => ["Save history-#{key}", "Save history-#{key}"])
```

Comment:
Can you please explain what's going on here?

Reply:
Certainly! The `changes` method, provided by `Mongoid::Changeable` (in v4.0 and later, `Mongoid::Dirty` before then) [since v2.4](https://github.com/mongodb/mongoid/commit/f3d84a84ee40baddbbd5b7f0c64be5271892d9bf) copies the API of `ActiveModel::Dirty#changes` and both return an `ActiveSupport::HashWithIndifferentAccess` where the keys are attribute names and the values are an array with the first item being the value before the change and last item being the value after the change.

The specs include models that are overriding the changes method with ones that violate that spec, because their are values in the Hash that are not arrays. [One instance](https://github.com/mongoid/mongoid-history/commit/49fda9145b9a886e93d4caca0d704b327b1177ab) appears to just copy the flaw from the other spec. The original test goes back to [the pull request that caused the slowdown](https://github.com/mongoid/mongoid-history/pull/158/files). The implementation has extra code to deal with the differences between the overridden changes method in the spec and the normal changes method used everywhere else in the suite.

```ruby
changes[k].class == Array ? changes[k].last : changes[k]
```

All of the other specs and documentation use or show the correct format. It's not as impactful as not calling `changes` once or multiple times, but it also contributed to the slowdown.